IdentityMicroService/Application/Services/AuthenticationService.cs

```diff
@@ -1,181 +1,223 @@
-        {
-            await _signInManager.SignOutAsync();
-        }
+        var res = await _signInManager.PasswordSignInAsync(user, userForAuthentication.Password, false, false);
+        if (res.Succeeded) return user;
+        return null;
+    }
 
-        public async Task<Account> CreateUserAsync(RegistrationUserDTO model)
-        {
-            var result = await _userManager.CreateAsync(new Account { Email = model.Email, UserName = model.Email }, model.Password);
-            if (!result.Succeeded)
-            {
-                return null;
-            }
-
-            var user = await _userManager.FindByEmailAsync(model.Email);
-            if (user == null)
-            {
-                return null;
-            }
-
-            var res = await _signInManager.PasswordSignInAsync(user, model.Password, false, false);
-            if (res.Succeeded)
-            {
-                return user;
-            }
+    public async Task SignOutAsync()
+    {
+        await _signInManager.SignOutAsync();
+    }
 
-            return null;
-        }
+    public async Task<Account> CreateUserAsync(RegistrationUserDto model)
+    {
+        var result = await _userManager.CreateAsync(new Account { Email = model.Email, UserName = model.Email },
+            model.Password);
+        if (!result.Succeeded) return null;
+
+        var user = await _userManager.FindByEmailAsync(model.Email);
+        if (user == null) return null;
+
+        var res = await _signInManager.PasswordSignInAsync(user, model.Password, false, false);
+        if (res.Succeeded) return user;
+
+        return null;
+    }
 
-        public async Task AddUserRoleAsync(Account user, UserRole role)
+    public async Task AddUserRoleAsync(Account user, UserRole role)
+    {
+        var roles = await _userManager.GetRolesAsync(user);
+        if (roles.Count > 0) await _userManager.RemoveFromRolesAsync(user, roles);
+
+        await _userManager.AddToRoleAsync(user, role.ToString());
+    }
+
+    public async Task<Account> GetUserById(string userId)
+    {
+        return await _userManager.FindByIdAsync(userId);
+    }
+
+    public async Task<bool> CreatePatientAsync(RegistrationUserDto model)
+    {
+        var transaction = await _dBContext.Database.BeginTransactionAsync();
+
+        try
         {
-            var roles = await _userManager.GetRolesAsync(user);
-            if (roles.Count > 0)
-            {
-                await _userManager.RemoveFromRolesAsync(user, roles);
-            }
+            var user = await CreateUserAsync(model);
 
-            await _userManager.AddToRoleAsync(user, role.ToString());
-        }
+            await AddUserRoleAsync(user, UserRole.Patient);
 
-        public async Task<Account> GetUserById(string userId)
+            await transaction.CommitAsync();
+        }
+        catch (Exception)
         {
-            return await _userManager.FindByIdAsync(userId);
+            return false;
```

Comment:
You need at least to write to console your exception, it'll be very hard to fix it